bench/00_transform/cub.cu

```diff
@@ -0,0 +1,80 @@
+#include "matx.h"
+#include <nvbench/nvbench.cuh>
+
+
+using namespace matx;
+
+//
+// types of data to run tests on
+//
+// using sort_types = nvbench::type_list<matxFp16Complex, cuda::std::complex<float>, cuda::std::complex<double>, float, double>;
+using sort_types = nvbench::type_list<double>;
+
+////////////////////////////////////////////////////////////////////////////////
+///
+///\brief benchmark for 1D Sorts
+///
+///\param param in
+///
+////////////////////////////////////////////////////////////////////////////////
+template <typename ValueType>
+void sort1d_range(
+                 nvbench::state &state,
+                 nvbench::type_list<ValueType>
+                 )
+{
+  const int dataSize = static_cast<int>(state.get_int64("Tensor Size"));
+
+  auto sortedData = matx::make_tensor<ValueType>({dataSize});
+  auto randomData = matx::make_tensor<ValueType>({dataSize});
+
+  randomGenerator_t < ValueType > randData(dataSize, 0);
+
+  auto randTensorView = randData.template GetTensorView<sortedData.Rank()>(sortedData.Shape(), NORMAL);
+
+  (randomData = randTensorView).run();
```

Comment:
You probably need a prefetch here before running the benchmark since it's managed memory

Reply:
prefetched both